test-data/unit/check-classes.test

```diff
@@ -1887,6 +1887,18 @@ b = a.bar
 [out]
 main:9: error: Incompatible types in assignment (expression has type "A", variable has type "B")
 
+[case testNestedGetAttr]
+def foo() -> object:
+    def __getattr__() -> None:  # no error because not in a class
+        pass
+    return __getattr__
+
+class X:
+    def foo(self) -> object:
+        def __getattr__() -> None:  # no error because not directly inside a class
+            pass
+        return __getattr__
+
```

Comment:
I would add  two more tests like this:
```python
def make_getattr_good() -> Callable[[str], int]: ...
__getattr__ = make_getattr_good()  # OK

def make_getattr_bad() -> Callable[[], int]: ...
__getattr__ = make_getattr_bad()  # E: Invalid signature...
```

Because this may be one of typical use cases for PEP 572. This will likely require to also call `check_getattr_method` from `visit_assignment_statement` (not only from function definition like now).

Reply:
Good point. I'll also check for `from other_module import __getattr__`.

Reply:
And another one I realized I don't cover in the code I just pushed: `from other_module import other_name as __getattr__`.